Exercise_2.py

```diff
@@ -1,23 +1,53 @@
-# Python program for implementation of Quicksort Sort 
-  
-# give you explanation for the approach
+"""
+Quick Sort:
+I am taking last value of the array as the pivot
+and rearranging all the values depending
+if they are greater or lesser than the pivot
+We will have a border value, everything to the left of the border will be smaller than the pivot
+is the current number in consideration smaller than the pivot, if yes we will swap the value with the border
+and then we advance both pointers i.e border and the current value in consideration.
+if the current number is not smaller than the pivot then we just advance the value pointer else we swap the value with
+the border and advance both pointers, when the list ends, we are going to swap the pivot with the border value, we will see
+that the value left of pivot will be smaller than the pivot and values to the right of pivot will be greater than the
+pivot. After partition we will be runnning the quick sort algorithm recursively and applyingquicksort on left list and right list
+ till low becomes >=high
+Time complexity: O(nsquared) worst, during first partition, one list has 1 element while rest has all the other elements
+Space Complexity: I can't figure out
+:return: Sorted Array
+Issue: Not running on leetcode
+"""
+import random
 def partition(arr,low,high):
-  
-  
-    #write your code here
-  
+
+    pivotindex=high
+    # print("pivotindexis", pivotindex)
+    border=low
+    for i in range(low, high+1):
+        if arr[i]<pivotindex:
```

Comment:
Change the pivotIndex to arr[pivotIndex]

Reply:
Thank you Manoj. I changed it. Now this is running fine. Hope if you could review my change of Excerise 2.